src/components/user-defined-sets/sets.js

```diff
@@ -0,0 +1,159 @@
+import { Set as ImmutableSet, OrderedMap as ImmutableOrderedMap } from 'immutable';
+
+// Class for storing Set objects of string IDs (cell IDs, gene IDs, etc...).
+// The collection of sets is stored as a map whose values are Set objects.
+// Think of the static methods here as reducer action functions.
+// Motivation for using reducer pattern:
+// https://medium.com/@dan_abramov/you-might-not-need-redux-be46360cf367
+
+export default class Sets {
+    static initialState = {
+      namedSets: ImmutableOrderedMap(),
+      currentSet: ImmutableSet(),
+    };
+
+    /**
+     * Modifies the set entry for a named set.
+     * @param {object} state The current state.
+     * @param {string} key The set key.
+     * @param {Set} set The set value.
+     * @returns {object} The new state.
+     */
+    static setNamedSet(state, key, set) {
+      return {
+        ...state,
+        namedSets: state.namedSets.set(key, ImmutableSet(set)),
+      };
+    }
+
+    /**
+     * Deletes the set entry for a named set.
+     * @param {object} state The current state.
+     * @param {string} key The set key.
+     * @returns {object} The new state.
+     */
+    static deleteNamedSet(state, key) {
+      if (state.namedSets.has(key)) {
+        return {
+          ...state,
+          namedSets: state.namedSets.delete(key),
+        };
+      }
+      return state;
+    }
+
+    /**
+     * Deletes set entries for all named sets.
+     * @param {object} state The current state.
+     * @returns {object} The new state.
+     */
+    static deleteAllNamedSets(state) {
```

Comment:
I'd be in favor of at least commenting these out until they are actually in use. (There is a tension here: If you're making a library, it would make sense to fill out a basic library of verbs at the start, but this feels a little different to me.)